libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -450,99 +445,83 @@ static inline int is_active(struct gic_dist_map *gic_dist, int irq, int vcpu_id)
     return is_spi_active(gic_dist, irq, vcpu_id);
 }
 
-static inline int vgic_add_overflow_cpu(struct lr_of *lr_overflow, struct virq_handle *irq)
+static inline int vgic_irq_enqueue(vgic_t *vgic, vm_vcpu_t *vcpu, struct virq_handle *irq)
```

Comment:
Let's have this return a bool then also, as we don't have any error codes.
```suggestion
static inline boot vgic_irq_enqueue(vgic_t *vgic, vm_vcpu_t *vcpu, struct virq_handle *irq)
```

Reply:
Done, also changed int to bool in many other places also.

Reply:
So based on our discussion about using the bool only for predicate-like function names, I changed this back to using int, zero for success and non-zero otherwise. Just for the sake of consistency.